packages/services/src/kernel/default.ts

```diff
@@ -466,8 +466,8 @@ export class KernelConnection implements Kernel.IKernelConnection {
     if (this.status === 'dead') {
       throw new Error('Kernel is dead');
     }
-    this._clearKernelState();
     this._updateStatus('restarting');
+    this._clearKernelState();
```

Comment:
I moved this line above the status change to be more consistent with other status changes. For example, on disposal we first update the status, *then* we clear the kernel state. This allows someone to monitor the state and possibly do some cleanup before we clean out everything associated with the kernel.